### PR TITLE
Gene exp filter

### PR DESCRIPTION
**Pull request by airenzp**
## Description

This PR adds to support to filter  by gene expression.

## Checklist

[Check each task](https://github.com/stjude/proteinpaint/wiki/Pull-Request-Checklist) that has been performed or verified to be not applicable.
- [x] Tests: added and/or passed unit and integration tests, or N/A
- [x] Todos: commented or documented, or N/A
- [x] Notable Changes: updated release.txt, prefixed a commit message with "fix:" or "feat:", added to an internal tracking document, or N/A

### CODE REVIEW

File: server/src/termdb.filter.js [hunk 1]
```diff
@@ -48,11 +48,9 @@ export async function getFilterCTEs(filter, ds, CTEname = 'f') {
...
 		} else if (!item.tvs) {
 			throw `filter item should have a 'tvs' or 'lst' property`
-		} else if (
-			item.tvs.term.id &&
-			(!item.tvs.term.type || !isNonDictionaryType(item.tvs.term.type)) &&
-			!ds.cohort.termdb.q.termjsonByOneid(item.tvs.term.id)
-		) {
+		} else if (isNonDictionaryType(item.tvs.term.type)) {
+			f = { CTEname: CTEname_i, CTEs: [`${CTEname_i} AS (select id as sample from sampleidmap)`], values: [] }
+		} else if (item.tvs.term.id && !ds.cohort.termdb.q.termjsonByOneid(item.tvs.term.id)) {
```
Comment (xzhou82) [inline server/src/termdb.filter.js:53]: i suppose it should be extended after line 69 to plug in geneExp type

Comment (airenzp) [reply server/src/termdb.filter.js:53]: This is just a fake filter handling so it does not break this code. Filtering by gene expression does not query the db. The filtering is being done in the getData.

File: server/src/termdb.matrix.js [hunk 2]
```diff
@@ -178,7 +186,14 @@ async function getSampleData(q) {
 				const values = data.gene2sample2value.get(tw.term.gene)
 				const value = Number(values[sampleId])
 				let key = value
-				if (tw.q?.mode == 'discrete') key = getBinLabel(tw.q, lst, value)
+				if (ranges.length) {
+					const filterBin = getBin(ranges, value)
+					if (filterBin == -1) continue
```
Comment (airenzp) [inline server/src/termdb.matrix.js:191]: See here, if the value does not fall in the filter range the sample it is not added

**Comment by airenzp** [comment]
Moving forward with this logic would mean that other methods that query the db, like get_samples, would use getData instead so they get the filtered samples for non dict terms. It is another major change that would affect a lot of code.

**Review by xzhou82 — APPROVED** [review]
thanks works, next:
1. cover geneExp in filter integration test
2. at filter UI, should indicate term type under a geneExp/geneVariant pill

**Pull request merged**